controllers/alert_controller.go

```diff
@@ -99,68 +124,32 @@ func (r *AlertReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *AlertReconciler) validate(ctx context.Context, alert v1beta1.Alert) error {
-	var provider v1beta1.Provider
-	providerName := types.NamespacedName{Namespace: alert.Namespace, Name: alert.Spec.ProviderRef.Name}
-	if err := r.Get(ctx, providerName, &provider); err != nil {
-		return fmt.Errorf("failed to get provider %s, error: %w", providerName.String(), err)
-	}
-
-	if !apimeta.IsStatusConditionTrue(provider.Status.Conditions, meta.ReadyCondition) {
-		return fmt.Errorf("provider %s is not ready", providerName.String())
-	}
+func (r *AlertReconciler) reconcile(ctx context.Context, obj *v1beta1.Alert) (ctrl.Result, error) {
+	// Mark the resource as under reconciliation
+	conditions.MarkReconciling(obj, meta.ProgressingReason, "")
 
-	return nil
-}
-
-func (r *AlertReconciler) recordSuspension(ctx context.Context, alert v1beta1.Alert) {
-	if r.MetricsRecorder == nil {
-		return
+	// validate alert spec and provider
+	if err := r.validate(ctx, obj); err != nil {
+		conditions.MarkFalse(obj, meta.ReadyCondition, meta.FailedReason, err.Error())
+		return ctrl.Result{Requeue: true}, err
```

Comment:
```suggestion
		return ctrl.Result{}, err
```

returning an error automatically results in a requeue.